kpet/data.py

```diff
@@ -314,17 +316,38 @@ def matches(self, target):
 
 class Suite(Object):    # pylint: disable=too-few-public-methods
     """Test suite"""
+
+    def validate_case_ids(self):
+        """
+        Check case IDs are unique.
+        """
+        id_set = set()
+        repeated_id_set = set()
+        for case in self.cases:
+            # pylint: disable=redefined-builtin, invalid-name
+            id = case.id
```

Comment:
Shadows id()

Reply:
Yeah, I knooow :(. I think naming a function "id" was a bad move by Python developers. Grabbing such a common name. That's why I told pylint to bugger off. However, if you'd prefer I didn't do this, I can rename them to "case_id" and "suite_id".

Reply:
Unintentionally shadowing anything is something I never do. Maybe that's a habit from C.
If you really want to, you can merge the code as-is. I'll leave that up to you.